### PR TITLE
JSAB2-67 Connect troops frontend to backend

### PR DESCRIPTION
**Pull request by GuJialu**

### CODE REVIEW

File: frontend/.env [hunk 1]
```diff
@@ -0,0 +1 @@
+SERVER_URL=10.72.161.24
```
Comment (hongtaoTang) [inline frontend/.env]: .env file should be ignored

Comment (evasimonyi) [reply frontend/.env]: dotenv is merged, you can use it now @GuJialu

File: frontend/src/Troop/actionCreator.js [hunk 2]
```diff
@@ -0,0 +1,22 @@
+import { SERVER_URL } from 'react-native-dotenv';
+
+export const FETCH_TROOPS_SUCCEED = 'fetchTroopsSucceed';
+export const FETCH_TROOPS_START = 'fetchTroopsStart';
+export const FETCH_TROOPS_ERROR = 'fetchTroopsError';
+
+export function fetchTroops() {
+  const URL = `http://${SERVER_URL}/kingdom/troops`;
```
Comment (HanqingZ) [inline frontend/src/Troop/actionCreator.js]: url should be outside of the function

**Comment by GuJialu** [comment]
To test, please add Troop component in Game/index.js

File: frontend/src/Troop/index.jsx [hunk 3]
```diff
@@ -13,21 +19,63 @@ const styles = StyleSheet.create({
...
+    return (
+      <Text>{fetchError.message}</Text>
+    );
+  }
+
+  if (isLoading) {
+    return (
+      <ActivityIndicator size="large" color="#10978b" />
+    );
+  }
```
Comment (evasimonyi) [inline frontend/src/Troop/index.jsx]: put this under the calculations please

Comment (GuJialu) [reply frontend/src/Troop/index.jsx]: The calculation result won't render if isloading. Is there any reason for this?

File: frontend/src/Troop/troopReducer.js [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+};
+
+export default function troop(state = initialState, action) {
+  switch (action.type) {
+    case FETCH_TROOPS_REQUEST:
+      return { ...state, isLoading: true };
+    case FETCH_TROOPS_SUCCESS:
+      return { ...state, troops: action.payload, isLoading: false };
+    case FETCH_TROOPS_FAILURE:
+      return { ...state, error: action.payload, isLoading: false };
```
Comment (evasimonyi) [inline frontend/src/Troop/troopReducer.js]:
```suggestion
      return { ...state, isLoading: false, error: action.payload };
```

please keep the order of these things from the initialState for readability
troops, isLoading, error

File: frontend/src/Troop/index.jsx [hunk 5]
```diff
@@ -13,21 +20,63 @@ const styles = StyleSheet.create({
...
+  const error = useSelector((state) => state.troop.error);
+  const dispatch = useDispatch();
+
+  useEffect(() => {
+    dispatch(fetchTroops());
+  }, []);
+
+  if (error) {
+    return (
+      <Text>{`${error.message} Oops`}</Text>
```
Comment (evasimonyi) [inline frontend/src/Troop/index.jsx]:
```suggestion
      <Text>{`Oops, ${error.message}`}</Text>
```

File: frontend/src/Troop/actionCreator.js [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+  return (dispatch) => {
+    dispatch({ type: FETCH_TROOPS_REQUEST });
+    fetch(URL)
+      .then((response) => {
+        if (response.status === 200) {
+          return response.json();
+        }
+        throw new Error(response.status);
+      })
+      .then((json) => dispatch({ type: FETCH_TROOPS_SUCCESS, payload: json.troops }))
```
Comment (evasimonyi) [inline frontend/src/Troop/actionCreator.js]: don't call it json please, you can call it `response` or some form of `response` :)

**Pull request merged**